src/Ocelot.Provider.Kubernetes/Kube.cs

```diff
@@ -35,17 +33,14 @@ public virtual async Task<List<Service>> GetAsync()
             .ResourceClient(client => new EndPointClientV1(client))
             .GetAsync(_configuration.KeyOfServiceInK8s, _configuration.KubeNamespace);
 
-        _services.Clear();
-        if (endpoint?.Subsets.Count != 0)
-        {
-            _services.AddRange(BuildServices(_configuration, endpoint));
-        }
-        else
+        if ((endpoint?.Subsets?.Count ?? 0) == 0)
         {
             _logger.LogWarning(() => $"K8s Namespace:{_configuration.KubeNamespace}, Service:{_configuration.KeyOfServiceInK8s}; Unable to use: it is invalid. Address must contain host only e.g. localhost and port must be greater than 0!");
+            return new(0);
         }
 
-        return _services;
+        return BuildServices(_configuration, endpoint)
+            .ToList();
```

Comment:
I'm uncertain, but it may be necessary to wrap this statement with a `lock`, assuming that the class is intended to act as a singleton.

Reply:
I don't think this is necessary because the builder simply transforms the EndpointsV1 model into an Ocelot Service.
I've changed to `.ToList();` but in general ToList and AddRange both try to cast to a ICollection and create an array of the required length